enhancements/network/ocp-on-smart-nics.md

```diff
@@ -0,0 +1,276 @@
+    things, the Smart NICs would be their own `MachineConfigPool`,
+    separate from the masters and workers.) This may require changes
+    to tolerations and node selectors on existing DaemonSets, both
+    within OCP and in third-party operators or customer workloads (and
+    it may turn out that this approach is unworkable for this reason).
+
+  - The Smart NICs could be part of a second cluster running in
+    parallel with the primary cluster, with the control plane of the
+    NIC cluster running as pods within the primary cluster.
+
+  - The Smart NICs could be part of a second cluster which is
+    completely separate from the primary cluster, with its own master
+    nodes. (In some external discussion of this approach, it was
+    assumed that the masters in this scenario would have to be on ARM
+    hosts, but this isn't required; you can have a mix of x86_64
+    master nodes and ARM Smart NIC "worker" nodes.)
+
+  - The Smart NICs could be part of a second cluster, where the
+    control planes for both clusters are running in a different
+    cluster, ROKS-style.
+
+The latter two approaches allow for more isolation between the hosts
+and the NICs, since potentially even the cluster administrators of the
+primary cluster might not be able to influence the NIC cluster. This
+would be especially important if additional primary-cluster security
+and monitoring functions were offloaded to the NICs in the future.
+
+### Initial Development With Beta BlueField-2 Hardware
+
+Although we will eventually want to use IPMI to control and provision
+the NICs, just like we do for the hosts with bare-metal IPI, this will
+not work with the pre-GA BlueField-2 cards that we currently have
+access to. Instead, we will have to communicate with the NICs over PCI
+from the host. The most future-compatible approach is probably to
+create an IPMI proxy that can run on each host, claiming its own
+separate IP address on the provisioning network, and then proxying
+commands to the NIC via the PCI "RShim" interface. (Thus, we will have
+to first install OCP on the node, then deploy a DaemonSet to the node
+to run the IPMI proxy, then install OCP on the NIC via the proxy.)
+
+Later on, it should be possible to remove the proxy and instead speak
+IPMI directly to the NIC's Baseboard Management Controller connected
+directly via the out-of-band BMC network port on the card.
+
+(One unresolved problem with the "RShim" approach is that based on the
+proof of concept prototypes, it does not seem to be possible to do a
+fully-unsupervised installation via that interface. At one point, user
+intervention over a serial console is required to pick the correct
+option from a boot menu whose contents cannot be reliably predicted in
+advance. We may have to just have the user ssh in at some point during
+the install process to deal with this.)
+
+This would explicitly be a temporary hack which would not ever be part
```

Comment:
Rshim may not be suitable for running productized deployment, I'd still consider it as a userful tool to be added for debugging. wdyt? For example, either via rhcos image or included in the debug container.